dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -370,9 +370,8 @@ else if ( !params.hasPeriods() && params.hasFilterPeriods() )
 
             columns.add( columnAndAlias.asSql() );
 
-            // does repeatable stage exist?
-            if ( queryItem.hasProgramStage() && queryItem.getProgramStage().getRepeatable()
-                && queryItem.hasRepeatableStageParams() )
+            // asked for row context if allowed and needed
+            if ( rowContextAllowedAndNeeded( params, queryItem ) )
```

Comment:
Would be nice to have a e2e test to prevent DHIS2-15199 to happen in the future.

Reply:
Done